src/input.ts

```diff
@@ -6,33 +6,44 @@ export type ZipFileDescription = {
   modDate: Date
   bytes: ReadableStream<Uint8Array> | Uint8Array | Promise<Uint8Array>
   crc?: number // will be computed later
+  isFile: true
 }
+export type ZipFolderDescription = {
+  modDate: Date
+  isFile: false
+}
+export type ZipEntryDescription = ZipFileDescription | ZipFolderDescription;
 
 /** The file name and modification date will be read from the input if it is a File or Response;
  * extra arguments can be given to override the input's metadata.
  * For other types of input, the `name` is required and `modDate` will default to *now*.
  * @param encodedName will be coerced to string, so… whatever
  * @param modDate should be a Date or timestamp or anything else that works in `new Date()`
  */
-export function normalizeInput(input: File | Response | BufferLike | StreamLike, modDate?: any): ZipFileDescription {
+export function normalizeInput(input: File | Response | BufferLike | StreamLike, modDate?: any): ZipFileDescription;
+export function normalizeInput(input: undefined, modDate?: any): ZipFolderDescription;
+export function normalizeInput(input?: File | Response | BufferLike | StreamLike, modDate?: any): ZipEntryDescription {
```

Comment:
I think you meant `ZipFolderDescription `

Reply:
I think not? Overloads for File and Folder are just above it, but this definition is for the function to work as expected. Or maybe I misunderstood how overloads works?